fakebook/src/app/service/profile.service.ts

```diff
@@ -21,12 +21,45 @@ export class ProfileService {
     }
   };
 
+  /*
+    endpoints:
+        api/profiles/
+      ✔ [GET]         /profiles/{email}
+          + if email is null, check for currently logged in user
+      ✔ [GET]         /profiles/selection/{emails}
+          + used to get a collection of profiles by their emails
+      ✔ [POST]        /profiles/
+          + used for creating users
+          + takes in a `ProfileApiModel` as the request body
+      ✔ [PUT][AUTH]   /profiles/
+          + only works for currently logged in user
+          + used for updating the current user
+      - [POST]        /profiles/upload
+          + uploads an image via a form (input type='file')
+  */
   public GetProfile(email: string): Observable<User> /* profile */{
 
     return this.http.get<User>(this.baseUrl + email, this.headers);
   }
 
-  public GetProfileWithNullRoute(): Observable<User> /* null route */ {
+  public GetProfiles(emails: string[]): Observable<User> /* profile */{
+    // make empty collection of profiles
+    // emails={abc, 123, }
+
+    return this.http.get<User>(this.baseUrl + 'selection/' + emails, this.headers);
+  }
+
+  public createProfile(profile: User): Observable<User> {
+
+    return this.http.post<User>(`${this.baseUrl}`, profile, this.headers);
+  }
+
+  public UpdateProfile(email: string, profile: User): Observable<User>
+  {
+    return this.http.put<User>(this.baseUrl + email, profile, this.headers);
+  }
+  public GetProfileWithNullRoute(): Observable<User> /* null route */
+  {
```

Comment:
this code is also in another PR. code should not be tangled into multiple PRs that are submitted at the same time, since it leads to duplicate fragmented code review and unnecessary merge conflicts.